In [37]:
import pandas as pd
import numpy as np 
import seaborn as sns 
import scipy.stats as st
from scipy.stats import f_oneway
from scipy.stats import ttest_ind

In [45]:
iris=sns.load_dataset("iris") # Usamos la base de datos iris almacenada en la libreria seaborn
iris.head()
df = iris.loc[:, ['sepal_length', 'sepal_width', 'petal_length', 'petal_width']]
df.head()

sepal_length  sepal_width  petal_length  petal_width
0           5.1          3.5           1.4          0.2
1           4.9          3.0           1.4          0.2
2           4.7          3.2           1.3          0.2
3           4.6          3.1           1.5          0.2
4           5.0          3.6           1.4          0.2

## Tipos de pruebas

### a) Distribución t de Student 

# $$ SE = \frac{\sigma}{\sqrt{N}} $$

In [49]:
se_length=df.std()['sepal_length']/np.sqrt(10)
se_width=df.std()['sepal_width']/np.sqrt(10)

# $$ SED = \sqrt{\frac{\sigma_1^2}{N_1} + \frac{\sigma_1^2}{N_1}} $$

In [53]:
sed=np.sqrt((se_length**2) + (se_width**2))

# $$ t = \frac{x_1-x_2}{SED} $$

In [60]:
t_stat=(df.mean()['sepal_length'] - df.mean()['sepal_width'])/sed
t_stat

9.414779428410265

In [64]:
p = ttest_ind(df['sepal_length'], df['sepal_width'])
p

TtestResult(statistic=36.463283934449116, pvalue=7.027919392009369e-112, df=298.0)

### El valor absoluto del valor t es mayor que el valor crítico, entonces rechazamos la hipótesis nula, concluimos que las distribuciones son diferentes, teniendo en cuenta que la hipotesis original era que las distribuciones eran iguales.

## b) Correlación y coeficiente de Pearson

In [69]:
# Estudiamos la correlación entre length y width
df['sepal_length'].corr(df['sepal_width'])

-0.11756978413300208

In [73]:
# A continuación, estudiamos la correlación de Pearson entre ambas variables
st.pearsonr(df['sepal_length'], df['sepal_width'])

PearsonRResult(statistic=-0.11756978413300204, pvalue=0.15189826071144746)

### Correlación debil o inexistente 

# Análisis de la varianza (ANOVA)

In [79]:
f_oneway(df['sepal_length'], df['sepal_width'])

F_onewayResult(statistic=1329.571075284254, pvalue=7.027919392009998e-112)

###F Test es 1,329.57 y su pvalue correspondiente es 7.027e-112. Dado que es un número inferior a 0.05 rechazamos la hipótesis nula de que las medias entre variables son similares y concluimos que sí hay diferencias